wallet-api/src/Wallet/Emulator/Types.hs

```diff
@@ -200,7 +200,7 @@ isValidated txn = do
 data Event n a where
     -- | An direct action performed by a wallet. Usually represents a "user action", as it is
     -- triggered externally.
-    WalletAction :: Wallet -> n () -> Event n [Tx]
+    WalletAction :: Wallet -> n b -> Event n ([Tx], Either WalletAPIError b)
```

Comment:
Hm, no, the signature is deliberately restricted here. WalletAPI errors don't affect the trace other than aborting the wallet action, which I'm pretty sure they'll already do. Why did you change this?

Reply:
... and wallet actions deliberately don't get to produce a value other than unit.

Reply:
In order for the HTTP server to return the correct thing using the Trace api, it will need to know the result of the wallet action. Why does `WalletAction` not return the type of the wallet action currently, it seems strange to throw away the result of the action? I could combine `WalletAPIError` and `AssertionError`? It seems strange that the error type of `MonadEmulator` is an `AssertionError` anyway. That way we would have `WalletAction :: Wallet -> n b -> Event n ([Tx], b)`.

Alternatively I can not use the Trace api for the http server?

Reply:
> and wallet actions deliberately don't get to produce a value other than unit.

but they do in the `WalletAPI`, why don't they here?